model/k8s/metrics.yaml

```diff
@@ -63,6 +63,21 @@ groups:
       - ref: network.interface.name
       - ref: network.io.direction
 
+  # k8s.container.* metrics
+  - id: metric.k8s.container.status.state
+    type: metric
+    metric_name: k8s.container.status.state
+    stability: experimental
+    brief: ""
+    instrument: updowncounter
+    unit: ""
+    attributes:
+      - ref: k8s.container.state
+        requirement_level: required
+      - ref: k8s.container.reason
+        requirement_level: required
```

Comment:
If we have it as `required`, the problem is that states which do not have a reason won't have it provided. Maybe that could be left "unknown" or just empty?

@open-telemetry/semconv-k8s-approvers @open-telemetry/specs-semconv-approvers thoughts?